src/cli/hook-utils/get-check-update-data.js

```diff
@@ -0,0 +1,313 @@
+    if (value) {
+      const current = versionMap[sdk].current || '';
+      let latest = '',
+        error = null;
+
+      try {
+        latest = await fetchLatestModuleVersion(sdk);
+      } catch (err) {
+        error = err;
+      }
+
+      const update = !!current && !!latest && current !== latest;
+      const breaking = hasBreakingChange(current, latest);
+
+      versionMap[sdk] = {
+        ...versionMap[sdk],
+        latest,
+        update,
+        breaking,
+        error,
+      };
+    }
+  }
+
+  return { versionMap, inaccessibleFiles };
+}
+
+/**
+ * Reads project dependencies - cycles through project dependency files, extracts
+ * the listed dependencies, and adds it in to the version map with version info.
+ * @param cwd the current working directory of the CLI project
+ */
+async function readProjectDependencies(cwd) {
+  const versionMap = {};
+  const { dependencyFiles, inaccessibleFiles } = await gatherDependencyFiles(
+    cwd
+  );
+
+  for (const fileName of dependencyFiles) {
+    try {
+      const fileJSON = await getJSON(`${cwd}/${fileName}`);
+      const fileDependencies = await extractDependencies(fileJSON, fileName);
+
+      // For each dependency found, compare to SDK-related dependency
+      // list and, if known, update the versionMap with version information
+      for (const [key, val] of fileDependencies) {
+        for (const sdk of SLACK_JSON_SDKS) {
+          if (key !== '' && key === sdk) {
+            versionMap[sdk] = {
+              name: sdk,
+              current: val.version,
+            };
+          }
+        }
+      }
+    } catch (err) {
+      inaccessibleFiles.push({ name: fileName, error: err });
+    }
+  }
+  return { versionMap, inaccessibleFiles };
+}
+
+/**
+ * Reads and parses JSON file - if it works, returns the file contents.
+ * If not, returns an empty object
+ * @param filePath the path of the file being read
+ */
+async function getJSON(filePath) {
+  let fileContents = {};
+  try {
+    if (fs.existsSync(filePath)) {
+      fileContents = JSON.parse(fs.readFileSync(filePath, 'utf8'));
+    } else {
+      throw new Error('Cannot find a file at path ' + filePath, {
+        cause: 'Cannot find a file at path ' + filePath,
+      });
+    }
+  } catch (err) {
+    throw new Error(err.message, { cause: err });
+  }
+  return fileContents;
+}
+
+/**
+ * Gathers all related dependency files for the CLI project (package.json).
+ * @param cwd the current working directory of the CLI project
+ */
+async function gatherDependencyFiles(cwd) {
+  const { jsonDepFiles, inaccessibleFiles } = await getJSONFiles(cwd);
+  const dependencyFiles = jsonDepFiles;
+  return { dependencyFiles, inaccessibleFiles };
+}
+
+/**
+ * Gets the needed files that contain dependency info (package.json).
+ * @param cwd the current working directory of the CLI project
+ */
+async function getJSONFiles(cwd) {
+  const jsonFiles = ['package.json'];
```

Comment:
I removed the `package-lock.json` specific logic since it didn't make the code that much simpler and the `package.json` use case takes care of that scenario (since `package.json` is required but `package-lock` is optional). If we decide to keep it `package.json` only in terms of dependency files, then I will get rid of the for loop in line 132 to simplify the code more!

Reply:
This has been updated! the loop + array combo is gone ✨